src/v/config/configuration.cc

```diff
@@ -2776,6 +2776,27 @@ configuration::configuration()
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       512,
       {.min = 1, .max = 2048})
+  , shard_balancing_on_core_count_change(
+      *this,
+      "shard_balancing_on_core_count_change",
+      "If enabled, and if after a restart the number of cores changes, "
```

Comment:
They are mostly interchangeable. To be very precise, "shard" is a seastar concept, referring to a compartmentalized part of the Redpanda application that runs as a single thread. Actually you can run Redpanda with fewer or more shards than CPU cores, but the most common and sensible configuration is a 1-to-1 mapping. That's why we say that they are interchangeable.

I've been mostly using the word "core" for user-visible stuff such as configuration property names (because it is a concept that is more familiar to users), but switching to "shard" for precise and/or internal wording. Let me know if it makes sense or if you want to change it.

Reply:
Though for consistency I should probably rename the properties to `core_balancing_*` then :)

Reply:
I think consistency is good. If "shard" is mostly an implementation detail and does not impact how customers should understand or use the property, it makes sense to stick to `core_`. Thank you!